.github/workflows/deploy.yml

```diff
@@ -0,0 +1,58 @@
+name: Deployment
+
+on:
+  # https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_dispatch
+  workflow_dispatch:
+    branch:
+      description: 'The branch to deploy'
+      default: 'main'
+      required: false
+      type: string
+
+env:
+  NODE_ENV: development
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  NETLIFY_TOKEN: ${{ secrets.NETLIFY_TOKEN }}
+  NETLIFY_SITE_ID: ${{ secrets.NETLIFY_SITE_ID }}
+  FIGMA_TOKEN: ${{ secrets.FIGMA_TOKEN }}
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    environment: production
+    steps:
+      - uses: actions/checkout@v3
+        ref: ${{ github.event.inputs.branch }}
+        # https://github.com/actions/checkout/blob/main/src/input-helper.ts#L103
+        submodules: 'TRUE'
+
+      - name: Setup
+        uses: actions/setup-node@v3
+        with:
+          node-version: '14'
+          cache: 'yarn'
+
+      # https://github.com/actions/setup-node/issues/182
+      - name: Install Yarn
+        run: npm install --global yarn
+
+      - name: Install
+        run: yarn
+
+      - name: Test
+        run: yarn test
+
+      - name: Build
+        env:
+          NODE_ENV: production
+        run: yarn build
+
+      - name: Deploy Production
+        if: ${{ github.ref == 'refs/heads/main' }}
+        # https://github.com/zendeskgarden/scripts/tree/main/src/netlify/deploy#command
+        run: yarn garden netlify-deploy public --production --message "https://github.com/$GITHUB_REPOSITORY/commit/$GITHUB_SHA"
+
+      - name: Deploy Draft
+        if: ${{ github.ref != 'refs/heads/main' }}
+        # https://github.com/zendeskgarden/scripts/tree/main/src/netlify/deploy#command
+        run: yarn garden netlify-deploy public --message "https://github.com/$GITHUB_REPOSITORY/commit/$GITHUB_SHA"
```

Comment:
Draft deploys using Garden's netlify token should always be connect with PR SHAs. See https://github.com/zendeskgarden/website/deployments for details. Garden developers are expected to use the `garden netlify-deploy` CLI and personal tokens for any one-off deployments (which should be very rare – draft deployments should be connected with PR discussions). This option confuses Garden's deployment-of-record between PRs and commits.

Reply:
Is this the only case for draft deployments? What would be an alternative here if we wanted to sign the message in a similar matter? Maybe by branch instead of commit to distinguish between deployment types..

Reply:
Ok, then I think we need to get to a place where this action matches the heuristic established by `deploy.js`. There are options for accomplishing this:
- enhance deploy.js to accept a `branch` parameter (see `react-components` [yarn new](https://github.com/zendeskgarden/react-components/blob/main/utils/scripts/new.js) or [yarn tag](https://github.com/zendeskgarden/react-components/blob/main/utils/scripts/tag.js) for the preferred `commander` setup)
- use the [`garden github-deploy`](https://github.com/zendeskgarden/scripts/tree/main/src/github/deploy#readme) CLI directly (this has the disadvantage of skipping the bandwidth check)

Reply:
Agreed. Since we decided to go with `scripts/deploy.js` offline, we can be sure all of that is ran. And since the `@actions/checkout` is pointing us to the right branch before installation, we can be assured it will be available in the `deploy.js` script.